### ISSUE TITLE
17.0.0 — Ambiguous use of 'overlay(alignment:content:)'

### ISSUE THREAD

**Issue by michaeldclifford** [urbanairship/ios-library#367]
# Preliminary Info

### What Airship dependencies are you using?

17.0.0

### What are the versions of any relevant development tools you are using?

Xcode 14.3.1, iOS Simulator 16.4

# Report

### What unexpected behavior are you seeing?

Build error: "Ambiguous use of 'overlay(alignment:content:)'"

### What is the expected behavior?

Successful build

### What are the steps to reproduce the unexpected behavior?

Update from Airship 16.12.1 to 17.0.0

### Do you have logging for the issue?

From the build logs:

```
.../Pods/Airship/Airship/AirshipCore/Source/StoryIndicator.swift:64:18:
    error: ambiguous use of 'overlay(alignment:content:)'
        .fill(indicatorColor(index))
        ^
.../Pods/Airship/Airship/AirshipCore/Source/ViewExtensions.swift:79:10:
    note: found this candidate
        func overlay<T: View>(
        ^
SwiftUI.View:3:28:
    note: found this candidate
        @inlinable public func overlay<V>(alignment: Alignment = .center, @ViewBuilder content: () -> V) -> some View where V : View
        ^

.../Pods/Airship/Airship/AirshipCore/Source/StoryIndicator.swift:64:18:
    Ambiguous use of 'overlay(alignment:content:)'
```

**Comment by khmMouna** [comment]
Thank you for reporting that, we expose an `overlay(alignment:content:)` that override the SwiftUI one. You can use the deprecated one as a workaround `overlay(_ overlay:, alignment: )` and we will work to fix it.

**Comment by michaeldclifford** [comment]
Thank you

FWIW, the use of `.overlay()` that it's objecting to is _also_ part of the Airship library, so I don't think we can necessarily use the workaround without unlocking the files and editing the Airship code (which we'd rather avoid).

But it's not a showstopper, as we'll just hold off on updating for now 🙂

**Comment by khmMouna** [comment]
Sorry for the misunderstood, I am not referring to the `. overlay`  that  is part of the Airship library. We don't recommend doing that. But, I think this error will appear if you are using `.overlay` in your app. It will conflit with the one in the Airship SDK.

**Comment by michaeldclifford** [comment]
We aren't currently using SwiftUI in our app, so I imagine this will affect all clients, whether they use `.overlay()` themselves or not.

**Comment by michaeldclifford** [comment]
Not a problem. Thank you both 🙂

**Comment by michaeldclifford** [comment]
Fixed in [17.0.1](https://github.com/urbanairship/ios-library/releases/tag/17.0.1) 👍
